### ISSUE TITLE
Fix false positives for values wrapped in parentheses in declaration-empty-line-before

### ISSUE THREAD

**Issue by ekumlin** [stylelint/stylelint#5639]
> Clearly describe the bug

In a SCSS file, if you have a rule whose value is surrounded in `()`, the next line will show as an `declaration-empty-line-before` error.

Works:
![image](https://user-images.githubusercontent.com/1350074/138538572-6526c8ed-603f-4e54-8042-f677a586dc45.png)

Does not work:
![image](https://user-images.githubusercontent.com/1350074/138538575-75207e27-bf18-43f7-a06f-1107f861bcd0.png)

> Which rule, if any, is the bug related to?

`declaration-empty-line-before`

> What code is needed to reproduce the bug?

```css
div {
  height: (100%);
  width: 100%;
}
```

This is a contrived example, but it also repros with more appropriate examples, such as:

```css
div {
  height: ($abc + 50px) * ($def + 80px);
  width: 100%;
}
```

> What Stylelint configuration is needed to reproduce the bug?

I don't think any specific configuration is needed as this rule is on by default. Here's what I'm using, with the irrelevant rules trimmed out:

```yml
customSyntax: postcss-scss
extends:
- stylelint-config-standard
plugins:
- stylelint-order
- stylelint-scss
rules:
  declaration-empty-line-before: "always"
```

> Which version of Stylelint are you using?

`14.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI: `stylelint "src/**/test.scss"`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to the capability of using `()` parentheses in SCSS, which can be used for order of operations math with variables.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
src/test.scss
 9:2  ✖  Expected empty line before declaration  declaration-empty-line-before
```

----

I did a quick search for this error in previous issues and didn't see any that brought up this particular issue.

It seems to be caused by [`isStandardSyntaxDeclaration:41`](https://github.com/stylelint/stylelint/blob/060310cfd6383175ee06fe4d15aee27a4e202671/lib/utils/isStandardSyntaxDeclaration.js#L41) expressly treating SCSS lines that start in `(` and end in `)` as "not a standard property declaration".

For example, in this file:

![image](https://user-images.githubusercontent.com/1350074/138539796-ea7fc46c-a2b4-4938-875e-c969e5070676.png)

The `height` property is not treated as a declaration:

![image](https://user-images.githubusercontent.com/1350074/138539801-e087b5bd-609b-433e-bbfc-66b501a95089.png)

I'm not sure what the expected use case of `isStandardSyntaxDeclaration:41` is, so I don't really want to mess with it &mdash; however, it seems to be causing these simple math parentheses to be ignored as declarations.
